homeassistant/components/hassio/auth.py

```diff
@@ -63,8 +63,8 @@ def _check_access(self, request: web.Request):
         """Check if this call is from Supervisor."""
         # Check caller IP
         hassio_ip = os.environ["HASSIO"].split(":")[0]
-        if request[KEY_REAL_IP] != ip_address(hassio_ip):
-            _LOGGER.error("Invalid auth request from %s", request[KEY_REAL_IP])
+        if ip_address(request.remote) != ip_address(hassio_ip):
```

Comment:
It would be nice if we can get here the real IP without respect the forward for proto

Reply:
`request[KEY_REAL_IP]` already respected the forward previously, so that didn't change.

It van be changed to:

```suggestion
        if ip_address(request.transport.get_extra_info("peername")[0]) != ip_address(hassio_ip):
```

That would use the connect peer from the transport layer directly.